src/datalogger.cpp

```diff
@@ -59,7 +59,8 @@ void Datalogger::sleepMCU(int milliseconds)
 
     reenableAllInterrupts(iser1, iser2, iser3);
 
-    offsetMillis += milliseconds; // systick was stopped, add slept milliseconds
+    //offsetMillis += milliseconds; // systick was stopped, add slept milliseconds
+   
```

Comment:
Why is this removed?  Since the systick was stopped during sleeping, don't we need to add these milliseconds back in?

Reply:
Ah, I'm not quite sure, but the line was causing issues. I thought maybe the sleepMCU function was being called in a different location or was affecting something else. It's not affecting the time calculation in the current code though. Maybe there's something different in the code that writes all of the output at once?

Reply:
it needs to be a subtraction, I had misunderstood the logic here. Applied a fix and can merge now.